api/v1/verticarestorepointsquery_types.go

```diff
@@ -0,0 +1,104 @@
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// VerticaRestorePointsQuerySpec defines the desired state of VerticaRestorePointsQuery
+type VerticaRestorePointsQuerySpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// +kubebuilder:validation:Required
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:text"
+	// The name of the VerticaDB CR that this VerticaRestorePointsQuery is defined for.  The
+	// VerticaDB object must exist in the same namespace as this object.
+	VerticaDBName string `json:"verticaDBName"`
+
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// Optional parameter that will limit the query to only restore points
+	// from this archvie
+	ArchiveName string `json:"archiveName"`
+}
+
+const (
+	ArchiveNm = "backup"
+)
+
+// VerticaRestorePointsQueryStatus defines the observed state of VerticaRestorePointsQuery
+type VerticaRestorePointsQueryStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// VerticaRestorePointsQuery is the Schema for the verticarestorepointsqueries API
+type VerticaRestorePointsQuery struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   VerticaRestorePointsQuerySpec   `json:"spec,omitempty"`
+	Status VerticaRestorePointsQueryStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// VerticaRestorePointsQueryList contains a list of VerticaRestorePointsQuery
+type VerticaRestorePointsQueryList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []VerticaRestorePointsQuery `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&VerticaRestorePointsQuery{}, &VerticaRestorePointsQueryList{})
+}
+
+func (vrqb *VerticaRestorePointsQuery) ExtractNamespacedName() types.NamespacedName {
+	return types.NamespacedName{
+		Name:      vrqb.ObjectMeta.Name,
+		Namespace: vrqb.ObjectMeta.Namespace,
+	}
+}
+
+func MakeSampleVrqbName() types.NamespacedName {
+	return types.NamespacedName{Name: "vrqb-sample", Namespace: "default"}
+}
+
+// MakeVrqb will make an VerticaRestorePointsQuery for test purposes
+func MakeVrqb() *VerticaRestorePointsQuery {
```

Comment:
What does `vrqb` stand for? I was thinking it should be `vrpq` for `VerticaRestorePointsQuery`